src/cpp/cpp_instantiate_template.cpp

```diff
@@ -289,7 +291,9 @@ const symbolt &cpp_typecheckt::instantiate_template(
 
 #ifdef DEBUG
   std::cout << "A: <";
-  forall_expr(it, specialization_template_args.arguments())
+  for(auto it = specialization_template_args.arguments().begin();
```

Comment:
At the risk of sounding ungrateful for the amazing clean up PRs... the ones that do still need iterators, I think they are best still with the macro.